src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasPointAndLineSlicer.java

```diff
@@ -182,29 +234,103 @@ private long createNewPointIdentifier(final PointIdentifierFactory pointIdentifi
     {
-        if (getCountries() != null && !getCountries().isEmpty())
+        Geometry currentTarget = target;
+        boolean fullyMatched = false;
+        // Start cut process
+        for (final org.locationtech.jts.geom.Polygon candidate : candidates)
         {
-            final String tagValue = tags.get(ISOCountryTag.KEY);
-            final String[] countryCodes = tagValue.split(ISOCountryTag.COUNTRY_DELIMITER);
-            for (final String countryCode : countryCodes)
+            Geometry clipped;
+            try
             {
-                // Break if any one the countries is inside the bound
-                if (getCountries().contains(countryCode))
+                clipped = currentTarget.intersection(candidate);
+            }
+            catch (final TopologyException exc)
+            {
+                logger.error(
+                        "Error while using regular intersection for line {}, attempting again with reduced precision",
+                        identifier, exc);
+                try
                 {
-                    return false;
+                    final GeometryPrecisionReducer precisionReducer = new GeometryPrecisionReducer(
+                            new PrecisionModel(CountryBoundaryMap.PRECISION_MODEL));
+                    precisionReducer.setPointwise(true);
+                    precisionReducer.setChangePrecisionModel(false);
+                    currentTarget = precisionReducer.reduce(target);
+                    clipped = currentTarget.intersection(candidate);
+                }
+                catch (final Exception newExc)
+                {
+                    logger.error(
+                            "Reduced precision still failed for line {}, rethrowing original exception",
+                            identifier, newExc);
+                    throw exc;
                 }
             }
 
-            // We've gone through all the countries and haven't seen one inside, it must be outside
-            return true;
+            // We don't want single point pieces
+            if (clipped.getNumPoints() > 1)
+            {
+                // Add to the results
+                final String countryCode = CountryBoundaryMap.getGeometryProperty(candidate,
+                        ISOCountryTag.KEY);
+                CountryBoundaryMap.setGeometryProperty(clipped, ISOCountryTag.KEY, countryCode);
+                addResult(clipped, results);
+
+                // Update target to be what's left after clipping
+                currentTarget = currentTarget.difference(candidate);
+                if (currentTarget.getDimension() == 1
+                        && currentTarget.getLength() < CountryBoundaryMap.LINE_BUFFER
+                        || currentTarget.getDimension() == 2
+                                && currentTarget.getArea() < CountryBoundaryMap.AREA_BUFFER
+                                && new DiscreteHausdorffDistance(currentTarget, candidate)
```

Comment:
This looks pretty interesting - worth a comment?

Reply:
good idea! I've also moved this out to a helper method, since that's a little easier to read and more reusable.